src/it/resources-noncompilable/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/InputRecordTypeParameterName.java

```diff
@@ -10,49 +10,51 @@
  * pattern = "(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"
  *
  */
-public record InputRecordTypeParameterName<t>(Integer x, String str) { // warn
+public record InputRecordTypeParameterName<t>(Integer x, String str) {
+    // violation above 'Record type name 't' must match pattern'
     public <TT> void foo() { }
 
-    <e_e> void foo(int i) {
+    <T> void foo(int i) {
     }
 }
 
-record Other <foo extends Serializable & Cloneable> // warn
+// violation below 'Record type name 'foo' must match pattern'
+record Other <foo extends Serializable & Cloneable>
 (LinkedHashMap<String, Node> linkedHashMap) {
 
     foo getOne() {
         return null;//comment
     }
 
-    <Tfo$o2T extends foo> /*comment*/Tfo$o2T getTwo(Tfo$o2T a) {
+    <T extends foo> /*comment*/T getTwo(T a) {
```

Comment:
same as above, looks like all is changed in this way

Reply:
I have replied you above, pls check that out